lib/improver/utilities/spatial.py

```diff
@@ -97,8 +97,11 @@ def convert_distance_into_number_of_grid_cells(
             which equates to the requested distance in the x and y direction.
         distance (Float):
             Distance in metres.
-        max_distance_in_grid_cells (int):
-            Maximum distance in grid cells.
+
+    Kwargs:
+        max_distance_in_grid_cells (int or None):
+            Maximum distance in grid cells.  Defaults to None, which bypasses
+            the check.
```

Comment:
Thanks for making the changes to this function. Just to complete this, it would be good to add a `Raises` section to the docstring, particularly, as this function raises a lot of different exceptions.

Reply:
I've done this but would like to propose some further changes.  The current "Raises" section looks like:
```    
Raises:
        ValueError:
            If the distance in grid cells is larger than the maximum dimension
            of the rectangular domain (measured across the diagonal).  Needed
            for neighbourhood processing.
        ValueError:
            If the distance in grid cells is zero.
        ValueError:
            If the distance in grid cells is negative.  (Assuming the distance
            argument is positive, this indicates one or more spatial axes are
            not correctly ordered.)
        Value Error:
            If max_distance_in_grid_cells is set and the distance in grid cells
            exceeds this value.  Needed for neighbourhood processing.
```
I suggest that:

- The first error might want a switch to turn it off, as it's only relevant for neighbourhood processing.  - The second error (distance of zero grid cells) really only merits a warning, doesn't it?

I'm reluctant to change these immediately as they may be quite entangled in neighbourhood processing, but would be tempted to add these suggestions to the rearrangement of spatial functions follow-on work discussed above.

Reply:
I think there's also another exception that's raised within the function: "Invalid grid: projection_x/y coords required", so we'd want to include that as well. 

Regarding your suggestions, yes, I wouldn't want to make any more functional changes now, as it might turn into a bit of a rabbit hole. This could be considered for follow-on work.